### PR TITLE
Increase code coverage in `spring-core` `serializer` package

### PR DESCRIPTION
**Pull request by devdilson**
Just a small contribution.
- Increase code coverage of classes in spring core serializer package.
image

- Adjust some methods to use `AutoCloseable` interface.

### CODE REVIEW

File: spring-core/src/main/java/org/springframework/core/serializer/DefaultDeserializer.java [hunk 1]
```diff
@@ -67,8 +67,7 @@ public DefaultDeserializer(@Nullable ClassLoader classLoader) {
 	@Override
 	@SuppressWarnings("resource")
 	public Object deserialize(InputStream inputStream) throws IOException {
-		ObjectInputStream objectInputStream = new ConfigurableObjectInputStream(inputStream, this.classLoader);
-		try {
+		try (var objectInputStream = new ConfigurableObjectInputStream(inputStream, this.classLoader)){
```
Comment (sbrannen) [inline spring-core/src/main/java/org/springframework/core/serializer/DefaultDeserializer.java:70]: We don't use `var` in Spring Framework.

Thus, please remove all usage of `var` in this PR.

File: spring-core/src/main/java/org/springframework/core/serializer/support/DeserializingConverter.java [hunk 2]
```diff
@@ -70,8 +70,7 @@ public DeserializingConverter(Deserializer<Object> deserializer) {
 
 	@Override
 	public Object convert(byte[] source) {
-		ByteArrayInputStream byteStream = new ByteArrayInputStream(source);
-		try {
+		try(var byteStream = new ByteArrayInputStream(source);) {
```
Comment (sbrannen) [inline spring-core/src/main/java/org/springframework/core/serializer/support/DeserializingConverter.java:73]: There is no technical reason to close a `ByteArrayInputStream` since its `close()` method is a no-op.

Therefore, please revert this change consistently within this PR.

File: spring-core/src/test/java/org/springframework/core/serializer/SerializationConverterTests.java [hunk 3]
```diff
@@ -37,42 +43,77 @@ class SerializationConverterTests {
...
 	}
 
 
 	class UnSerializable implements Serializable {
 
 		private static final long serialVersionUID = 1L;
 
-		@SuppressWarnings({"unused", "serial"})
-		private Object object;
+		@SuppressWarnings({"unused", "serial"}) private Object object;
```
Comment (sbrannen) [inline spring-core/src/test/java/org/springframework/core/serializer/SerializationConverterTests.java:116]: Please revert this change.

File: spring-core/src/test/java/org/springframework/core/serializer/SerializerTests.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
```
Comment (sbrannen) [inline spring-core/src/test/java/org/springframework/core/serializer/SerializerTests.java:2]:
```suggestion
 * Copyright 2002-2023 the original author or authors.
```

Please ensure all created/updated files have the correct copyright year.

File: spring-core/src/test/java/org/springframework/core/serializer/SerializerTests.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+
+class SerializerTests {
+
+	private static final String SPRING_FRAMEWORK = "Spring Framework";
+
+
+	class SpyStringSerializer<T> implements Serializer<T> {
```
Comment (sbrannen) [inline spring-core/src/test/java/org/springframework/core/serializer/SerializerTests.java:36]: These helper classes should be `static` and placed at the end of the test class, after test methods.

**Review by sbrannen — CHANGES_REQUESTED** [review]
I've requested a few minor changes.

File: spring-core/src/main/java/org/springframework/core/serializer/DefaultDeserializer.java (hunk 1 above)
Comment (devdilson) [reply spring-core/src/main/java/org/springframework/core/serializer/DefaultDeserializer.java]: Good to know, removed all instances.

File: spring-core/src/main/java/org/springframework/core/serializer/support/DeserializingConverter.java (hunk 2 above)
Comment (devdilson) [reply spring-core/src/main/java/org/springframework/core/serializer/support/DeserializingConverter.java]: Reverted

File: spring-core/src/test/java/org/springframework/core/serializer/SerializationConverterTests.java (hunk 3 above)
Comment (devdilson) [reply spring-core/src/test/java/org/springframework/core/serializer/SerializationConverterTests.java]: Reverted

File: spring-core/src/test/java/org/springframework/core/serializer/SerializerTests.java (hunk 4 above)
Comment (devdilson) [reply spring-core/src/test/java/org/springframework/core/serializer/SerializerTests.java]: Added the changed.

File: spring-core/src/test/java/org/springframework/core/serializer/SerializerTests.java (hunk 5 above)
Comment (devdilson) [reply spring-core/src/test/java/org/springframework/core/serializer/SerializerTests.java]: Done

**Comment by devdilson** [comment]
I did the improvements @sbrannen thanks for reviewing it!

**Review by sbrannen — APPROVED** [review]

**Comment by sbrannen** [comment]
> Adjust some methods to use `AutoCloseable` interface.

Please note that this change is inappropriate according to the official contract of `Deserializer#deserialize(InputStream)`:

> Note: Implementations should not close the given InputStream (or any decorators of that InputStream) but rather leave this up to the caller.

However, there's no need to update this PR: I'll remove that change when merging.

**Comment by sbrannen** [comment]
This has been merged into `main` in 4979d8fded09cefb1b77c91abae506ff4f4a58f5 and revised in bfeca4a0bf0f02f06e42413065b5cc651ae94963.

Thanks
